File: Semester 4/Advanced Programming/ClientServerApplication/ApplicationServices/src/main/java/app/services/IObserver.java
package app.services;

import app.model.Cursa;
import app.model.Rezervare;

import java.util.List;

public interface IObserver {
    void seatsDecremented(Cursa cursa) throws ServiceException;
    void newBookings(List<Rezervare> rezervari) throws ServiceException;
}

File: Semester 4/Advanced Programming/RpcProtocolC#/Services/IObserver.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Domain;
namespace services
{
    
    public interface IObserver
    {
        void seatsDecremented(Cursa cursa);
        void newBookings(List<Rezervare> rezervari);
    }
}
